.github/workflows/ci.yml

```diff
@@ -27,13 +27,6 @@ jobs:
           go-version: ${{ env.DEFAULT_GO_VERSION }}
           check-latest: true
           cache-dependency-path: "**/go.sum"
-      - name: Tools cache
-        uses: actions/cache@v4
-        env:
-          cache-name: go-tools-cache
-        with:
-          path: .tools
-          key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('./internal/tools/**') }}
```

Comment:
Removing this means the tools will be built every run. That will slow down our CI system.

The cache is hashed on the file contents of `./internal/tools/`, meaning they will be updated whenever there is a tool update.

It does not seem needed, or desired, to remove this.

Reply:
Correct. Tomorrow, I will make a new PR with alternative solution (as mentioned in PR description). Closing this PR.

Reply:
Hmm. I thought the makefile changes could have been merged. They looked like good changes to included.

I'll look forward to your alternate then I guess.

Reply:
> They looked like good changes to included.

I thought so as well 😢 Take notice that I have decided to remove the cache after I saw it is useless as a separate commit.

> It does not give any value after the changes as make would rebuild most of the tools anyway. See: https://github.com/open-telemetry/opentelemetry-go/actions/runs/8717700613/job/23913426182.

I will debug it first. My guess is that the `go.mod` gets the "timestamp" of the time when `git checkout` happened and not when it was last modified according to `git history`.

Reopening as maybe I am missing something and more 👀 may help.

Reply:
```
rpajak@LAPTOP-K6FOL8LO:~/repos$ mkdir test
rpajak@LAPTOP-K6FOL8LO:~/repos$ cd test
rpajak@LAPTOP-K6FOL8LO:~/repos/test$ git clone https://github.com/open-telemetry/opentelemetry-go.git
Cloning into 'opentelemetry-go'...
remote: Enumerating objects: 42522, done.
remote: Counting objects: 100% (2196/2196), done.
remote: Compressing objects: 100% (475/475), done.
remote: Total 42522 (delta 1931), reused 1750 (delta 1720), pack-reused 40326
Receiving objects: 100% (42522/42522), 17.67 MiB | 1.67 MiB/s, done.
Resolving deltas: 100% (29225/29225), done.
rpajak@LAPTOP-K6FOL8LO:~/repos/test$ cd opentelemetry-go/
rpajak@LAPTOP-K6FOL8LO:~/repos/test/opentelemetry-go$ ls -la --time-style=full-iso
total 476
drwxr-xr-x 18 rpajak rpajak   4096 2024-04-18 09:50:12.252128594 +0200 .
drwxr-xr-x  3 rpajak rpajak   4096 2024-04-18 09:49:59.951545582 +0200 ..
-rw-r--r--  1 rpajak rpajak     42 2024-04-18 09:50:12.192125746 +0200 .codespellignore
-rw-r--r--  1 rpajak rpajak    252 2024-04-18 09:50:12.192125746 +0200 .codespellrc
drwxr-xr-x  8 rpajak rpajak   4096 2024-04-18 09:50:12.262129069 +0200 .git
-rw-r--r--  1 rpajak rpajak     89 2024-04-18 09:50:12.192125746 +0200 .gitattributes
drwxr-xr-x  4 rpajak rpajak   4096 2024-04-18 09:50:12.192125746 +0200 .github
-rw-r--r--  1 rpajak rpajak    309 2024-04-18 09:50:12.192125746 +0200 .gitignore
-rw-r--r--  1 rpajak rpajak    147 2024-04-18 09:50:12.192125746 +0200 .gitmodules
-rw-r--r--  1 rpajak rpajak  11486 2024-04-18 09:50:12.192125746 +0200 .golangci.yml
-rw-r--r--  1 rpajak rpajak    298 2024-04-18 09:50:12.192125746 +0200 .lycheeignore
-rw-r--r--  1 rpajak rpajak    309 2024-04-18 09:50:12.192125746 +0200 .markdownlint.yaml
-rw-r--r--  1 rpajak rpajak 208566 2024-04-18 09:50:12.192125746 +0200 CHANGELOG.md
-rw-r--r--  1 rpajak rpajak    543 2024-04-18 09:50:12.192125746 +0200 CODEOWNERS
-rw-r--r--  1 rpajak rpajak  21960 2024-04-18 09:50:12.192125746 +0200 CONTRIBUTING.md
-rw-r--r--  1 rpajak rpajak  11357 2024-04-18 09:50:12.192125746 +0200 LICENSE
...
```

Reply:
We could do something like: https://stackoverflow.com/a/55609950/859275

To touch the `go.mod`. I will try it.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/5220/commits/0841a31a1bff9219f10392c4f045ac2cd4be5730